### PR TITLE
Round settings

### LINKED ISSUE #29: Set up basic RoundController
**Description**
System that checks the time of a "round" of gameplay, starts and ends a round, sets gamestate on "game over", "victory" etc.
Send info about waves of vikings on to VikingManager depending on difficulty etc.

**Subtasks**
- [x] Sends object containing data to VikingManager for each wave.
- [x] Creates objects storing data for each wave of Vikings.
- [ ] Sets gamestate to game over/win etc.

### PR DESCRIPTION
**Pull request by MrJackey**
Closes #29 

**Description**  
The roundcontroller now scales difficulty depending the amount of players and current round.

**List of changes**  
- Created objects to control scaling
- The roundcontroller controls how vikings should spawn alongside their behaviour

**Additional context**  
Note that the scaling methods are not optimal and should not be used later on. A lower/upper limit should be implemented

### CODE REVIEW

File: Assets/00_Content/Scripts/Rounds/RoundController.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+				Debug.Assert(false, "Roundcontroller have access to a vikingController");
+				return;
+			}
+
+			ScalingData difficulty = playerDifficulties[PlayerManager.Instance && PlayerManager.Instance.NumPlayers > 0
+				? PlayerManager.Instance.NumPlayers - 1
+				: 0];
+
+			vikingController.SpawnDelay = difficulty.ScaledSpawnDelay(currentRound);
+			vikingController.StatScaling = new VikingScaling(difficulty, currentRound);
```
Comment (smeas) [inline Assets/00_Content/Scripts/Rounds/RoundController.cs:47]: When i read the code for the VikingController I thought that these properties were never changed from their defaults. Would having a method like `VikingController.SetDifficulty(ScalingData)` and letting the controller write to the properties instead make it clearer?

**Review by smeas — COMMENTED** [review]
Just some thoughts.

File: Assets/00_Content/Scripts/Rounds/RoundController.cs (hunk 1 above)
Comment (MrJackey) [reply Assets/00_Content/Scripts/Rounds/RoundController.cs:47]: I can totally see how that would improve clarity! Good idea!

File: Assets/00_Content/Scripts/Vikings/VikingStats.cs [hunk 2]
```diff
@@ -1,24 +1,28 @@
...
 
 		private float mood;
 		public float Mood => mood;
 
-		public VikingStats(VikingData data) {
-			this.data = data;
-			mood = data.startMood;
+		private float moodDeclineRate;
+
+		public VikingStats(VikingData data, VikingScaling scaling) {
```
[the remark this answers is not in the document]
Comment (MrJackey) [reply Assets/00_Content/Scripts/Vikings/VikingStats.cs:14]: The data object only contains the base values. The saved values are scaled based on the number of players and which round is currently being played.

**Review by tzaarela — APPROVED** [review]
Nothing to comment on, looks good 👍

**Review by smeas — APPROVED** [review]
Great work!

**Pull request merged**